airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py

```diff
@@ -226,14 +226,15 @@ def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs: Any) -
             ...]
         """
 
-        today = pendulum.now().date()
+        end_date = pendulum.now().date()
         start_date = pendulum.parse(self.start_date).date()
         if stream_state:
             prev_end_date = pendulum.parse(stream_state.get(self.cursor_field)).date()
-            start_date = prev_end_date.add(days=1)
-        end_date = today
-        if start_date > end_date:
-            return [None]
+            start_date = prev_end_date.add(days=1)  # do not include previous `end_date`
+            # always resync 2 previous days to be sure data is golden
+            # https://support.google.com/analytics/answer/1070983?hl=en#DataProcessingLatency&zippy=%2Cin-this-article
+            # https://github.com/airbytehq/airbyte/issues/12013#issuecomment-1111255503
+            start_date = start_date.subtract(days=2)
```

Comment:
Why do you need two `start_date` variables, if you redefine it?

Reply:
You mean line 233? just to make it obvious. I guess `start_date.subtract(days=1)` would be not quiet clear. Instead, I added on more line of code but the reader will have less questions

Or line 230? It will not be redefined if `stream_state` is None